### PR TITLE
Optimize preprocessor directives on Solaris for uv__tcp_keepalive

### PR DESCRIPTION
**Pull request by panjf2000**
Reduce the amount of code being compiled and trim trailing whitespace in passing.

### CODE REVIEW

**Comment by panjf2000** [comment]
Sorry for not being aware of this improvement in my previous PR, but let's "fix" it now.

File: src/unix/tcp.c [hunk 1]
```diff
@@ -499,35 +500,36 @@ int uv__tcp_keepalive(int fd, int on, unsigned int delay) {
...
+  /* Note that the consequent probes will not be sent at equal intervals on Solaris,
    * but will be sent using the exponential backoff algorithm. */
   intvl = idle/3;
   cnt = 3;
   int time_to_abort = intvl * cnt;
   if (setsockopt(fd, IPPROTO_TCP, TCP_KEEPALIVE_ABORT_THRESHOLD, &time_to_abort, sizeof(time_to_abort)))
     return UV__ERR(errno);
+#endif
 
   return 0;
```
[the remark this answers is not in the document]
Comment (panjf2000) [reply src/unix/tcp.c]: I've pushed a new commit, please check out if it's what you wanted.

File: src/unix/tcp.c [hunk 2]
```diff
@@ -499,43 +500,44 @@ int uv__tcp_keepalive(int fd, int on, unsigned int delay) {
...
   intvl = idle/3;
   cnt = 3;
   int time_to_abort = intvl * cnt;
   if (setsockopt(fd, IPPROTO_TCP, TCP_KEEPALIVE_ABORT_THRESHOLD, &time_to_abort, sizeof(time_to_abort)))
     return UV__ERR(errno);
-
-  return 0;
 #endif
 
+#else
```
Comment (bnoordhuis) [inline src/unix/tcp.c:534]:
```suggestion
#endif

#else  /* !defined(__sun) */
```

File: src/unix/tcp.c [hunk 3]
```diff
@@ -552,6 +554,7 @@ int uv__tcp_keepalive(int fd, int on, unsigned int delay) {
     return UV__ERR(errno);
 #endif
 
+#endif
```
Comment (bnoordhuis) [inline src/unix/tcp.c:557]:
```suggestion
#endif  /* !defined(__sun) */
```

**Review by bnoordhuis — APPROVED** [review]

**Comment by panjf2000** [comment]
Hi @bnoordhuis, should we move this forward?

**Review by bnoordhuis — APPROVED** [review]
LGTM, thanks Andy.

**Pull request merged**